internal/db/schema/migrations/oss/postgres/83/01_iam_role_grant_scope.up.sql

```diff
@@ -0,0 +1,173 @@
+-- Copyright (c) HashiCorp, Inc.
+-- SPDX-License-Identifier: MPL-2.0
+
+begin;
+
+  -- This table is a join table to map roles to the various scopes the role
+  -- grants apply to. role_id is a foreign key, but scope_id_or_special is not a
+  -- foreign key to scopes because we support specific special values which are
+  -- checked for in the constraint. We never query by scope_id_or_special so the
+  -- ordering of the primary key should be correct.
+  create table iam_role_grant_scope (
+    create_time wt_timestamp,
+    role_id wt_role_id not null -- pk
+      references iam_role(public_id)
+      on delete cascade
+      on update cascade,
+    scope_id_or_special text not null -- pk
+      constraint scope_id_or_special_is_valid
+      check (
+        length(trim(scope_id_or_special)) = 12
+          or
+        scope_id_or_special in ('global', 'this', 'children', 'descendants')
+      ),
+    primary key(role_id, scope_id_or_special)
+  );
+
+  insert into oplog_ticket (name, version)
+    values
+    ('iam_role_grant_scope', 1);
+  
```

Comment:
Can you check this file for trailing whitespaces?

Reply:
I think that was the only place, but will fix.